File: sdk/csharp/src/Buildkite.Sdk/WaitStep.cs
namespace Buildkite.Sdk
{
    public class WaitStep : PipelineStep
    {
        public string? Wait { get; set; }
        public bool? ContinueOnFailure { get; set; }

        public WaitStep()
        {
        }

        public WaitStep(string label)
        {
            Label = label;
        }
    }
}

File: sdk/typescript/src/waitStep.ts
import { DependsOn } from "./types";

export interface WaitStep {
    allow_dependency_failure?: boolean;
    branches?: string[] | string;
    continue_on_failure?: boolean;
    depends_on?: Array<DependsOn | string> | null | string;
    id?: string;
    identifier?: string;
    if?: string;
    key?: string;
    label?: string;
    name?: string;
    wait: string;
}
